hasura/graphql-engine
# Add export database schema option to the settings page on console  (close #2371)

Add export database schema option to the settings page on console  (close #2371)


### Description
console now support DB schema export, which allows the user to download database schema using [pg_dump](https://hasura.io/docs/1.0/graphql/manual/api-reference/pgdump.html#id1) api.

### Changelog


### Affected components


### Related Issues
close #2371 

### Solution and Design
> pg_dump API is used to fetch the data and the data is further downloaded as `schema.sql` file.

### Steps to test and verify
1. go to settings page
2. click on `Export Schema` button 
3. user should be able to download a schema.sql file.
![image](https://user-images.githubusercontent.com/8408875/87897375-c3496c80-ca68-11ea-8dab-946718d7b7f0.png)

#### Catalog upgrade

Does this PR change Hasura Catalog version?

#### Metadata
Does this PR add a new Metadata feature?

#### GraphQL

#### Breaking changes
